libcxx/include/string_view

```diff
@@ -776,7 +776,35 @@ template <ranges::contiguous_range _Range>
 #endif
 
 // [string.view.comparison]
+
+#if _LIBCPP_STD_VER >= 20
+
```

Comment:
Let's regroup all the C++ >= 20 definitions into the same block. There's a bunch of those below. There's also conditions that can be simplified now, e.g. some `#if _LIBCPP_STD_VER < 20` that would be made tautological since they'd be in the `#else`.

Reply:
I pushed a wrong commit in the branch. It's fixed now.